platform/src/main/java/org/stellar/anchor/platform/config/PropertyClientsConfig.java

```diff
@@ -1,156 +1,60 @@
-  Map<String, String> signingKeyToClientNameMap = null;
-
-  @PostConstruct
-  public void postConstruct() {
-    // In favor of migrating to signingKeys and domains, copy signingKey to signingKeys and domain
-    // to domains if signingKeys and domains are not set.
-    clients.forEach(
-        clientConfig -> {
-          if (!isEmpty(clientConfig.getSigningKey())) {
-            clientConfig.setSigningKeys(setOf(clientConfig.getSigningKey()));
-          }
-          if (!isEmpty(clientConfig.getDomain())) {
-            clientConfig.setDomains(setOf(clientConfig.getDomain()));
-          }
-        });
-  }
-
-  @Override
-  public ClientsConfig.ClientConfig getClientConfigBySigningKey(String signingKey) {
-    if (signingKeyToClientNameMap == null) {
-      signingKeyToClientNameMap = Maps.newHashMap();
-      clients.forEach(
-          clientConfig -> {
-            if (clientConfig.getSigningKeys() != null && !clientConfig.getSigningKeys().isEmpty()) {
-              for (String key : clientConfig.getSigningKeys()) {
-                signingKeyToClientNameMap.put(key, clientConfig.getName());
-              }
-            }
-          });
-    }
-    return getClientConfigByName(signingKeyToClientNameMap.get(signingKey));
-  }
-
-  @Override
-  public ClientConfig getClientConfigByDomain(String domain) {
-    if (domainToClientNameMap == null) {
-      domainToClientNameMap = Maps.newHashMap();
-      clients.forEach(
-          clientConfig -> {
-            if (clientConfig.getDomains() != null && !clientConfig.getDomains().isEmpty()) {
-              for (String domainName : clientConfig.getDomains()) {
-                domainToClientNameMap.put(domainName, clientConfig.getName());
-              }
-            }
-          });
-    }
-    return getClientConfigByName(domainToClientNameMap.get(domain));
-  }
-
-  public ClientConfig getClientConfigByName(String name) {
-    if (clientMap == null) {
-      clientMap = Maps.newHashMap();
-      clients.forEach(clientConfig -> clientMap.put(clientConfig.getName(), clientConfig));
-    }
-    return clientMap.get(name);
-  }
+  ClientsConfigType type;
+  String value;
 
   @Override
   public boolean supports(@NotNull Class<?> clazz) {
-    return PropertyClientsConfig.class.isAssignableFrom(clazz);
+    return ClientsConfig.class.isAssignableFrom(clazz);
   }
 
   @Override
   public void validate(@NotNull Object target, @NotNull Errors errors) {
-    PropertyClientsConfig configs = (PropertyClientsConfig) target;
-    configs.clients.forEach(clientConfig -> validateClient(clientConfig, errors));
-  }
-
-  private void validateClient(ClientConfig clientConfig, Errors errors) {
-    debugF("Validating client {}", clientConfig);
-    if (isEmpty(clientConfig.getName())) {
-      errors.reject("empty-client-name", "The client.name cannot be empty and must be defined");
+    // TODO(jiahui): ANCHOR-622 Support more types of clients config
```

Comment:
What other types do we plan to support beyond `json` and `yaml`?

Reply:
This was a reminder for myself during implementation, will remove